server/restapi/restapi.coffee

```diff
@@ -0,0 +1,59 @@
+Api = new Restivus
+	useDefaultAuth: true
+	prettyJson: true
+
+
+Api.addCollection ChatMessage
+Api.addCollection ChatRoom
+
+
+Api.addRoute 'version', authRequired: false,
+	get: ->
+		version = {api: '0.1', rocketchat: '0.5'}
+		status: 'success', versions: version
+
+Api.addRoute 'publicRooms', authRequired: true,
+	get: ->
+		rooms = ChatRoom.find({ t: 'c' }, { sort: { msgs:-1 } }).fetch()
+		if rooms
+			status: 'success', rooms: rooms
+		else
+			statusCode: 404
+			body: status: 'fail', message: 'Rooms not found'
+
+# join a room
+Api.addRoute 'rooms/:id/join', authRequired: true,
+	post: ->
+		Meteor.runAsUser this.userId, () =>
+			Meteor.call('joinRoom', @urlParams.id)
+		status: 'success'   # need to handle error
+
+
+# leave a room
+Api.addRoute 'rooms/:id/leave', authRequired: true,
+	post: ->
+		Meteor.runAsUser this.userId, () =>
+			Meteor.call('leaveRoom', @urlParams.id)
+		status: 'success'   # need to handle error
+
+
+# get messages in a room
+Api.addRoute 'rooms/:id/messages', authRequired: true,
+	get: ->
```

Comment:
OT:  but REST-full API really *SCREAMS*  regression tests.  Can we add a 'test' tree somewhere in src?   I think it will be useful even if they are not automated right-the-way.